alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/Mixer.scala

```diff
@@ -11,6 +11,7 @@ import shapeless.Lazy
 trait Mixer[A, B] {
   def mix(a: A): B
   def inject(b: B, a: A): A
+  def modify(f: B => B): A => A = a => inject(f(mix(a)), a)
```

Comment:
I'd rather this took an `A` as well. In principle the function `A=>A` floating around your code is meaningless and it would be easy to forget what it does